[FIX] mail: discuss notification settings properly handle long text

alexkuhn opened the pull request:
Before this commit, the notification settings in discuss app that is used to mute channels was not showing items properly in some languages like vietnamese.

This happens because the width is hard-coded to 150px as to keep the dropdown menu small.

This commit fixes the issue by putting a max-width of 250px for dropdown menu and sub-menu, so that it keeps right positioning of submenu. If text is too long, it know wraps.

![Screenshot 2024-05-24 at 12 06 08](https://github.com/odoo/odoo/assets/6569390/4d727151-2a49-4a20-bfcb-4be4940efc7b)

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/167017).

tsm-odoo reviewed (commented):
Missing space between title and arrow in case of overflow. `flex-grow` is good when there is no overflow but a margin should be added as well :)

![image](https://github.com/odoo/odoo/assets/48757558/95a1fe92-9461-4a91-a9f8-1b58e2e29b4a)

alexkuhn commented:
> Missing space between title and arrow in case of overflow. `flex-grow` is good when there is no overflow but a margin should be added as well :)

Fixed (I added `ms-2` on arrow similarly to `17.2`'s PR)

tsm-odoo reviewed (approved):
@robodoo r+

alexkuhn commented:
Cleaning of `<button><button/><button/>` is made in `17.2`. `<Dropdown/>` are different in `17.0` and `17.2`, so better keep as it is. Tests have been adapted to specify the proper `button.dropdown-toggle`

@robodoo r+
@fw-bot up to saas-17.1
reason: dedicate pr for 17.2: https://github.com/odoo/odoo/pull/166673
